### PR TITLE
Add metro e2e tests

### PR DESCRIPTION
**Pull request by lukeatdell**
# Description
Adding e2e tests for powerstore metro replication.
The test suit tests:
- The metro volume is mounted with multiple device mount paths on the Kubernetes worker node for the provisioned csi persistent volume (PV).
- Multiple host paths exist between the Kubernetes worker node and the PowerStore array for each side of the metro volume.
- All host paths are in 'active ready' state.
- The persistent volume is writable.
- The persistent volume continues to be writable after taking down all host paths for one of the two PowerStore arrays in the metro replication session.

Additional info:
- Updated kubernetes e2e test module versions and necessary implementations.
- Relocated k8s e2e tests under csi-powerstore/test/e2e/k8s

# GitHub Issues

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1443 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Backward compatibility is not broken

# How Has This Been Tested?

- [x] Metro e2e test executed
image

### CODE REVIEW

File: tests/e2e/driver/resources/sc.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: powerstore-metro
+parameters:
+  arrayID: "unique" # PowerStore Global ID
+  replication.storage.dell.com/ignoreNamespaces: "false"
```
Comment (santhoshatdell) [inline tests/e2e/driver/resources/sc.yaml:7]: _ignoreNamespaces_ parameter is not needed.

File: tests/e2e/driver/resources/sc.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  replication.storage.dell.com/isReplicationEnabled: "true"
+  replication.storage.dell.com/mode: METRO
+  replication.storage.dell.com/remoteSystem: "system-ID" # PowerStore system name
```
Comment (santhoshatdell) [inline tests/e2e/driver/resources/sc.yaml:10]: Remote PowerStore system name

File: tests/e2e/driver/resources/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Copyright © 2020-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (santhoshatdell) [inline tests/e2e/driver/resources/deployment.yaml:1]: 2024

File: tests/e2e/driver/resources/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+# To test the driver just run from root directory of repository
+# > kubectl create -f ./tests/simple/
+#
+# You can find all resources in testpowerstore namespace.
+# Check if pod is created and Ready and Running by running
+# > kubectl get all -n testpowerstore
+# (if it's in CrashLoopback state then driver installation wasn't successful, check logs of node and controller)
+# After that can go into created container and check if everything is mounted correctly.
+#
+# After that you can uninstall the testing PVCs and StatefulSet
```
Comment (santhoshatdell) [inline tests/e2e/driver/resources/deployment.yaml:29]: Some of these non-applicable comments can be removed.

**Review by donatwork — DISMISSED** [review]

**Review by santhoshatdell — DISMISSED** [review]

**Review by mbasha-dell — APPROVED** [review]

**Review by lukeatdell — COMMENTED** [review]
LGTM.

**Review by santhoshatdell — APPROVED** [review]

**Review by mbasha-dell — APPROVED** [review]

**Pull request merged**